File: apps/backend/app/models/todo.py
"""
Todo model.
"""
from datetime import datetime
from sqlalchemy import Column, Integer, String, Text, DateTime, Enum, ForeignKey
import enum

from app.models.base import Base


class Priority(enum.Enum):
    LOW = "LOW"
    MEDIUM = "MEDIUM"
    HIGH = "HIGH"


class Todo(Base):
    __tablename__ = "todos"

    id = Column(Integer, primary_key=True, index=True)
    user_id = Column(String, nullable=False, index=True)
    task = Column(Text, nullable=False)
    priority = Column(Enum(Priority), default=Priority.MEDIUM, nullable=False)
    due_date = Column(DateTime, nullable=True)
    journal_entry_id = Column(Integer, ForeignKey("journal_entries.id"), nullable=True)
    created_at = Column(DateTime, default=datetime.utcnow, nullable=False)
    updated_at = Column(DateTime, default=datetime.utcnow, nullable=False)

File: apps/web/lib/types/todo.ts
export type Priority = "LOW" | "MEDIUM" | "HIGH";

export interface Todo {
  id: number;
  user_id: string;
  task: string;
  priority: Priority;
  due_date: string | null;
  journal_entry_id?: number | null;
  created_at: string;
  updated_at: string;
}

export interface CreateTodoRequest {
  task: string;
  priority?: Priority;
  due_date?: string | null;
  journal_entry_id?: number | null;
  user_id?: string;
}